File: Java/leetcode75/src/test/java/org/phoenix/leetcode/challenges/Problem32_MaximumTwinSumOfALinkedListTest.java
package org.phoenix.leetcode.challenges;

import org.junit.jupiter.api.Test;
import org.phoenix.leetcode.challenges.Problem29_DeleteTheMiddleNodeOfALinkedList.ListNode;

import static org.junit.jupiter.api.Assertions.assertEquals;

class Problem32_MaximumTwinSumOfALinkedListTest {

    private final Problem32_MaximumTwinSumOfALinkedList test = new Problem32_MaximumTwinSumOfALinkedList();

    @Test
    void pairSum() {
        ListNode head = new ListNode(5);
        head.next = new ListNode(4);
        head.next.next = new ListNode(2);
        head.next.next.next = new ListNode(1);

        assertEquals(6, test.pairSum(head));
    }
}

File: JavaScript/leetcode75/src/test/Problem32_MaximumTwinSumOfALinkedList.test.js
const pairSum = require("../main/Problem32_MaximumTwinSumOfALinkedList");

function ListNode(val, next) {
  this.val = val === undefined ? 0 : val;
  this.next = next === undefined ? null : next;
}

test("should return max twin sum", () => {
  let list = new ListNode(5, 4);
  list.next = new ListNode(4, 2);
  list.next.next = new ListNode(2, 1);
  expect(pairSum(list)).toBe(6);
});
